### ISSUE TITLE
XdgSurface does not support field access

### ISSUE THREAD

**Issue by willruggiano** [swaywm/zig-wlroots#52]
Ok, trying to follow up on #51.

I am trying to build tinywl in a fresh Docker environment, since I am on NixOS and I am suspicious that something therein is causing problems.

When I try to build tinywl, I get the following error;
```
root@943d453e7a51:/src/tinywl# zig build
./tinywl.zig:180:70: error: type '?*.wlroots.types.xdg_shell.XdgSurface' does not suppo
rt field access
                const parent_node = @intToPtr(?*wlr.SceneNode, parent.data) orelse {
                                                                     ^
./tinywl.zig:228:32: error: type '?*.wlroots.types.xdg_shell.XdgSurface' does not suppo
rt field access
                _ = xdg_surface.role_data.toplevel.setActivated(false);
                               ^
tinywl...The following command exited with error code 1:
/zig-linux-x86_64-0.9.1/zig build-exe /src/tinywl/tinywl.zig -lc -lwayland-server -lxkb
common -lwlroots -lpixman-1 -cflags -std=c99 -- /src/tinywl/zig-cache/zig-wayland/wayla
nd-protocol.c -cflags -std=c99 -- /src/tinywl/zig-cache/zig-wayland/xdg-shell-protocol.
c --cache-dir /src/tinywl/zig-cache --global-cache-dir /root/.cache/zig --name tinywl -
-pkg-begin wayland /src/tinywl/zig-cache/zig-wayland/wayland.zig --pkg-end --pkg-begin 
xkbcommon /src/tinywl/deps/zig-xkbcommon/src/xkbcommon.zig --pkg-end --pkg-begin wlroot
s /src/src/wlroots.zig --pkg-begin wayland /src/tinywl/zig-cache/zig-wayland/wayland.zi
g --pkg-end --pkg-begin xkbcommon /src/tinywl/deps/zig-xkbcommon/src/xkbcommon.zig --pk
g-end --pkg-begin pixman /src/tinywl/deps/zig-pixman/pixman.zig --pkg-end --pkg-end -I 
/usr/include/libdrm -I /usr/include/pixman-1 -I /usr/include/pixman-1 --enable-cache 
error: the following build command failed with exit code 1:
/src/tinywl/zig-cache/o/273e1967212913bf98518fdd2053d313/build /zig-linux-x86_64-0.9.1/
zig /src/tinywl /src/tinywl/zig-cache /root/.cache/zig
```

This is the Dockerfile;
```docker
FROM ubuntu:latest

RUN apt update -y
RUN apt-get install -y git wget xz-utils pkg-config clang libpixman-1-dev libwayland-dev libwayland-bin libwlroots-dev

RUN wget https://ziglang.org/download/0.9.1/zig-linux-x86_64-0.9.1.tar.xz
RUN tar -xf zig-linux-x86_64-0.9.1.tar.xz
ENV PATH="${PATH}:/zig-linux-x86_64-0.9.1"

RUN git clone https://github.com/swaywm/zig-wlroots /src
WORKDIR /src
RUN git submodule update --init
```

`apt list --installed` shows the following packages. Perhaps one of them is not the correct version? (just spitballing)
```
...
clang-14/jammy,now 1:14.0.0-1ubuntu1 amd64 [installed,automatic]
...
libpixman-1-0/jammy,now 0.40.0-1build4 amd64 [installed,automatic]
libpixman-1-dev/jammy,now 0.40.0-1build4 amd64 [installed]
...
libwayland-bin/jammy,now 1.20.0-1 amd64 [installed]
libwayland-client0/jammy,now 1.20.0-1 amd64 [installed,automatic]
libwayland-cursor0/jammy,now 1.20.0-1 amd64 [installed,automatic]
libwayland-dev/jammy,now 1.20.0-1 amd64 [installed]
libwayland-egl1/jammy,now 1.20.0-1 amd64 [installed,automatic]
libwayland-server0/jammy,now 1.20.0-1 amd64 [installed,automatic]
...
libwlroots-dev/jammy,now 0.15.1-2 amd64 [installed]
libwlroots10/jammy,now 0.15.1-2 amd64 [installed,automatic]
...
libxkbcommon-dev/jammy,now 1.4.0-1 amd64 [installed,automatic]
...
linux-libc-dev/jammy-updates,jammy-security,now 5.15.0-43.46 amd64 [installed,automatic]
...
llvm-14-dev/jammy,now 1:14.0.0-1ubuntu1 amd64 [installed,automatic]
llvm-14-linker-tools/jammy,now 1:14.0.0-1ubuntu1 amd64 [installed,automatic]
llvm-14-runtime/jammy,now 1:14.0.0-1ubuntu1 amd64 [installed,automatic]
llvm-14-tools/jammy,now 1:14.0.0-1ubuntu1 amd64 [installed,automatic]
llvm-14/jammy,now 1:14.0.0-1ubuntu1 amd64 [installed,automatic]
...
wayland-protocols/jammy,now 1.25-1 all [installed,automatic]
```

**Comment by novakne** [comment]
hmm yeah it seems that 06297208176ac530b4882c92d6b064bda467c001 is the problem since the return type is now an optional, that is a simple fix, I'll open a PR

**Comment by ifreund** [comment]
Should be fixed by e749df37374ffc7fe5b558239519cdec44fdf717, thanks for the bug report!

**Comment by willruggiano** [comment]
Thanks!
